### DISCUSSION TITLE
Integrate mermaid charts

### DISCUSSION THREAD

**Discussion by jamesburton** [KDE/ghostwriter#843] [Ideas]
Given that GitHub and many other git providers support mermaid charts within their markdown it becomes more important to support this format to be a viable choice of standalone editor.

**Comment by Potherca** [comment]
I was wondering the same thing for PlantUML. In either case, I was wondering how it would look in the UI?

**Comment by dnc77** [comment]
Is there support for mermaid, in any way with ghostwriter?

**Comment by mrunderhill89** [comment]
This is probably a bigger thing to ask for than it looks. Mermaid, to my knowledge, is a Javascript library, so you'd need to either port it to C++ and Qt to make it work with what Ghostwriter already has, or implement some kind of web renderer with Javascript support, at which point you're basically reinventing Electron. Web-based programs like GitHub's editor get away with it because they're already written in Javascript, so integrating Mermaid is easy for them. Not so much for a native desktop app.

**Reply by dnc77** [reply]
Hi there,

Thank you for your response. I found out that as Ghostwriter uses pandoc to render and also offers the feature to pass extra parameters that one can simply put '-Fmermaid-filter' in the extra parameters and Ghostwriter will tell Pandoc to filter the output through mermaid.

That will work just fine.

**Reply by amadeus84** [reply]
@dnc77 

Do you mind elaborating a bit on putting '-Fmermaid-filter' in the extra parameters? I'm running ghostwriter 25.04.2 and I have pandoc 3.1.11.1 installed. Upon startup, ghostwriter says:

[     0.104 663430 ghostwriter ] WARNING   No such plugin for spec "pandoc"
[     0.113 663430 ghostwriter ] INFO      Backup files will be stored in "/home/amadeus/.local/share/ghostwriter/backups/"
[     0.126 663430 ghostwriter ] INFO      Using pandoc version 3.1.11.1
[     0.126 663430 ghostwriter ] INFO      Using multimarkdown version 6.7.0
[     0.127 663430 ghostwriter ] INFO      Using cmark version 0.30.3

Thanks!
